src/essreflectometry/orso.py

```diff
@@ -1,15 +1,173 @@
 # SPDX-License-Identifier: BSD-3-Clause
 # Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
-import warnings
+"""ORSO utilities for reflectometry."""
+import os
+import platform
+from datetime import datetime, timezone
+from typing import NewType, Optional
 
+import sciline
+from dateutil.parser import parse as parse_datetime
+from orsopy.fileio import base as orso_base
+from orsopy.fileio import data_source, orso, reduction
 
-def not_found_warning():
+from .types import Filename, RawData, Reference, Run, Sample
+
+
+class OrsoExperiment(
+    sciline.Scope[Run, data_source.Experiment], data_source.Experiment
+):
+    """ORSO experiment for a run."""
+
+
+class OrsoInstrument(
+    sciline.Scope[Run, data_source.InstrumentSettings], data_source.InstrumentSettings
+):
+    """ORSO instrument settings for a run."""
+
+
+class OrsoOwner(sciline.Scope[Run, orso_base.Person], orso_base.Person):
+    """ORSO owner of a file."""
+
+
+class OrsoReduction(sciline.Scope[Run, reduction.Reduction], reduction.Reduction):
+    """ORSO measurement for a run."""
+
+
+class OrsoSample(sciline.Scope[Run, data_source.Sample], data_source.Sample):
+    """ORSO sample of a run."""
+
+
+OrsoCreator = NewType('OrsoCreator', orso_base.Person)
+"""ORSO creator, that is, the person who processed the data."""
+
+OrsoIofQDataset = NewType('OrsoIofQDataset', orso.OrsoDataset)
+"""ORSO dataset for reduced I-of-Q data."""
+
+OrsoDataSource = NewType('OrsoDataSource', data_source.DataSource)
+"""ORSO data source."""
+
+OrsoMeasurement = NewType('OrsoMeasurement', data_source.Measurement)
+"""ORSO measurement."""
+
+
+def parse_orso_experiment(raw_data: RawData[Run]) -> OrsoExperiment[Run]:
+    """Parse ORSO experiment metadata from raw NeXus data."""
+    return OrsoExperiment(
+        data_source.Experiment(
+            title=raw_data['title'],
+            instrument=raw_data['name'],
```

Comment:
The `name` appears to be in NXinstrument, not NXentry: https://manual.nexusformat.org/classes/base_classes/NXinstrument.html#nxinstrument-name-field

Reply:
So we do need to make this Amor-specific because the files don't follow the standard. 9https://github.com/scipp/essreflectometry/pull/27#discussion_r1477839389)

Reply:
Actually, in our test file, the name shows up in both places